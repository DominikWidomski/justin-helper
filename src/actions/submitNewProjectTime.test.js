const mockStdin = require('mock-stdin');
const keys = require('./utils/keys');
const mockStdoutWrite = require('./utils/mockStdout');
const submitNewProjectTime = require("./submitNewProjectTime");

describe("actions", () => {
    describe("submitNewProjectTime", () => {
        test("default", async () => {
            const stdin = mockStdin.stdin();
            const stdoutMock = mockStdoutWrite();
            const postHandler = jest.fn((url, { data }) => {
                return {
                    data: {
                        attributes: {
                            ...data.attributes,
                            name: "Project name"
                        }
                    }
                };
            });
            
            const client = {
                post: postHandler
            }
            
            const projectTime = {
                user_id: '1',
                project_id: '1',
                date: '2018-01-01',
                duration_mins: 420
            };

            const projects = {
                data: [{
                    id: '1',
                    attributes: {
                        name: 'project',
                    }
                }]
            };
            
            process.nextTick(() => {
                stdin.send('Y');
                stdin.send(keys.enter);
            });
            
            const answer = await submitNewProjectTime(client, '2018-01-02', projectTime, 1, projects);

            expect(postHandler.mock.calls).toMatchSnapshot();
            expect(stdoutMock.flush()).toMatchSnapshot();
            expect(answer).toMatchSnapshot();

            stdoutMock.restore();
        })
    })
})