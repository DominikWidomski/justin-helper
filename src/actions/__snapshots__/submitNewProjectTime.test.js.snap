// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions submitNewProjectTime default 1`] = `
Array [
  Array [
    "/project-times",
    Object {
      "data": Object {
        "attributes": Object {
          "date": "2018-01-02",
          "duration_mins": 420,
          "project_id": "1",
          "user_id": 1,
        },
        "type": "project-times",
      },
    },
  ],
]
`;

exports[`actions submitNewProjectTime default 2`] = `
Array [
  "",
  "? About to submit 7h to \\"project\\" on Tuesday 2018-01-02 (y/N) ",
  "",
  "",
  "",
  "? About to submit 7h to \\"project\\" on Tuesday 2018-01-02 (y/N) Y",
  "",
  "",
  "",
  "? About to submit 7h to \\"project\\" on Tuesday 2018-01-02 Yes",
  "",
  "",
  "
",
  "Submitted 7 to project on Tuesday 2018-01-02",
]
`;

exports[`actions submitNewProjectTime default 3`] = `
Object {
  "attributes": Object {
    "date": "2018-01-02",
    "duration_mins": 420,
    "name": "Project name",
    "project_id": "1",
    "user_id": 1,
  },
}
`;
